src/pages/permissions/panels/CreateTLSIdentityPanel.tsx

```diff
@@ -0,0 +1,151 @@
+import { useNotify } from "@canonical/react-components";
+import SidePanel from "components/SidePanel";
+import type { FC } from "react";
+import usePanelParams from "util/usePanelParams";
+import * as Yup from "yup";
+import { useFormik } from "formik";
+import NotificationRow from "components/NotificationRow";
+import type { TLSIdentityFormValues } from "../forms/NameWithGroupForm";
+import GroupSelection from "./GroupSelection";
+import useEditHistory from "util/useEditHistory";
+import GroupSelectionActions from "../actions/GroupSelectionActions";
+import { useGroups } from "context/useGroups";
+import { createFineGrainedTlsIdentity } from "api/auth-identities";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { base64EncodeObject } from "util/helpers";
+import NameWithGroupForm from "../forms/NameWithGroupForm";
+
+interface GroupEditHistory {
+  groupsAdded: Set<string>;
+}
+
+interface Props {
+  onSuccess: (identityName: string, token: string) => void;
+}
+
+const CreateTLSIdentityPanel: FC<Props> = ({ onSuccess }) => {
+  const panelParams = usePanelParams();
+  const notify = useNotify();
+  const queryClient = useQueryClient();
+
+  const { data: groups = [], error, isLoading } = useGroups();
+
+  const {
+    desiredState,
+    save: saveToPanelHistory,
+    undo: undoMappingChanges,
+  } = useEditHistory<GroupEditHistory>({
+    initialState: {
+      groupsAdded: new Set(),
+    },
+  });
+
+  if (error) {
+    notify.failure("Loading panel details failed", error);
+  }
+
+  const modifyGroups = (newGroups: string[], isUnselectAll?: boolean) => {
+    if (isUnselectAll) {
+      saveToPanelHistory({
+        groupsAdded: new Set(),
+      });
+    } else {
+      saveToPanelHistory({
+        groupsAdded: new Set(newGroups),
+      });
+    }
+  };
+
+  const closePanel = () => {
+    panelParams.clear();
+    notify.clear();
+  };
+
+  const handleSubmit = (values: TLSIdentityFormValues) => {
+    closePanel();
```

Comment:
This is too early to close the panel. We should do this only below in the `then` branch. Otherwise, in the error case the user input will be lost. By postponing the panel close, the user can resubmit with the previous inputs.